### PR TITLE
Add Metadata in std::os::fortanix_sgx::io::FromRawFd

### PR DESCRIPTION
**Pull request by mzohreva**
Needed for https://github.com/fortanix/rust-sgx/pull/291

cc @jethrogb

### CODE REVIEW

File: library/std/src/sys/sgx/ext/io.rs [hunk 1]
```diff
@@ -71,18 +74,38 @@ impl AsRawFd for net::TcpListener {
     }
 }
 
+/// Metadata for `TcpStream`.
+#[derive(Debug, Clone, Default)]
+pub struct TcpStreamMetadata {
+    /// Local address of the TCP stream
+    pub local_addr: Option<String>,
```
Comment (shirshak55) [inline library/std/src/sys/sgx/ext/io.rs:81]: shouldn't it be SocketAddr instead of String?

Comment (jethrogb) [reply library/std/src/sys/sgx/ext/io.rs:81]: No, see https://docs.rs/fortanix-sgx-abi/0.3.3/fortanix_sgx_abi/struct.Usercalls.html#networking

File: library/std/src/sys/sgx/ext/io.rs [hunk 2]
```diff
@@ -71,18 +74,38 @@ impl AsRawFd for net::TcpListener {
     }
 }
 
+/// Metadata for `TcpStream`.
+#[derive(Debug, Clone, Default)]
+pub struct TcpStreamMetadata {
```
Comment (jethrogb) [inline library/std/src/sys/sgx/ext/io.rs:79]: Missing stability attribute

File: library/std/src/sys/sgx/ext/io.rs [hunk 3]
```diff
@@ -71,18 +74,38 @@ impl AsRawFd for net::TcpListener {
...
-        let socket = sys::net::Socket::from_inner(fd);
-        net::TcpStream::from_inner(sys::net::TcpStream::from_inner((socket, None)))
+        let socket = sys::net::Socket::from_inner((fd, metadata.local_addr));
+        net::TcpStream::from_inner(sys::net::TcpStream::from_inner((socket, metadata.peer_addr)))
     }
 }
 
+/// Metadata for `TcpListener`.
+#[derive(Debug, Clone, Default)]
+pub struct TcpListenerMetadata {
```
Comment (jethrogb) [inline library/std/src/sys/sgx/ext/io.rs:98]: Missing stability attribute

Comment (mzohreva) [reply library/std/src/sys/sgx/ext/io.rs:100]: Fixed

**Review by jethrogb — APPROVED** [review]
LGTM

**Pull request merged**